input_tool.py

```diff
@@ -0,0 +1,203 @@
+    cur.execute("DROP TABLE IF EXISTS observations CASCADE")
+    cur.execute("DROP TABLE IF EXISTS variables CASCADE")
+
+def load_dataset_to_db(cur):
+    with open("dataset.json", 'rb') as dataset:
+        sample_dataset = json.load(dataset)
+    for owner in sample_dataset['owners']:
+        add_user(cur, owner)
+    for experiment in sample_dataset['experiments']:
+        add_experiment(cur, experiment)
+    for plot in (sample_dataset['plots']):
+        add_plot(cur, plot, sample_dataset['observation_dirs'])
+
+def get_observation_name(obs_dirs, filename):
+    for observation in obs_dirs:
+        obs_dir_path = os.path.join(EXPERIMENT_DATA_PATH, observation)
+        for plot in os.listdir(obs_dir_path):
+            if filename == plot:
+                return observation
+    return None
+
+def add_user(cur, user_obj):
+    # check preconditions
+    if 'owner_id' not in user_obj or 'username' not in user_obj:
+        print("Skipping current user: needs both owner_id and username")
+        return 1
+    elif 'first_name' not in user_obj or 'last_name' not in user_obj:
+        cur.execute("INSERT INTO owners (owner_id, username, first_name, last_name) VALUES (%s, %s, %s, %s)", (user_obj["owner_id"], user_obj["username"], user_obj["first_name"], user_obj["last_name"]))
+    else:
+        cur.execute("INSERT INTO owners (owner_id, username, first_name, last_name) VALUES (%s, %s, %s, %s)", (user_obj["owner_id"], user_obj["username"], user_obj["first_name"], user_obj["last_name"]))
+    return 0
+
+def delete_user(cur, username):
+    cur.execute("DELETE FROM owners WHERE username=%s", (username,))
+
+def add_experiment(cur, experiment_obj):
+    if 'experiment_id' not in experiment_obj or 'owner_id' not in experiment_obj:
+        print("Skipping current experiment: must provide both experiment_id and owner_id")
+        return 1
+    else:
+        cur.execute("INSERT INTO experiments (experiment_id, experiment_name, owner_id) VALUES (%s, %s, %s)", (experiment_obj["experiment_id"], experiment_obj["experiment_name"], experiment_obj["owner_id"]))
+    return 0
+
+def add_plot(cur, plot_obj, observation_dirs):
+    if 'plot_id' not in plot_obj or 'plot_file' not in plot_obj or 'experiment_id' not in plot_obj or 'group_id' not in plot_obj or 'observation_id' not in plot_obj:
+        print("Skipping current plot: needs plot, experiment, group, and observation ids")
+        return 1
+    
+    plot_filename = plot_obj['plot_file']
+    observation_name = get_observation_name(observation_dirs, plot_filename)
+    plot_file_path = os.path.join(EXPERIMENT_DATA_PATH, observation_name, plot_filename)
+
+    with open(plot_file_path, 'rb') as file:
+        dictionary = pickle.load(file)
+
+    # extract the div and script components
+    div = dictionary['div']
+    script = dictionary['script']
+
+    # parse filename
+    filename_no_extension = os.path.splitext(plot_filename)[0]
+    plot_components = filename_no_extension.split("_")
```

Comment:
Please put a comment here so the reader immediately knows what attribute each plot_component corresponds to